### DISCUSSION TITLE
Using id instead of uuid

### DISCUSSION THREAD

**Discussion by DriverCat** [spatie/laravel-event-sourcing#431] [Q&A]
Hi, I found an old topic on this theme for the parent package: https://github.com/spatie/laravel-event-projector/issues/73
But for the latest versions of this package i see a lot of mentions and relations for the `uuid` and `aggregate_uuid` fields. As far as I see, it is not possible now to use for example a primaryKey `id` instead of `uuid` field without changing package files. Is it true? Do anybody has such working cases?

**Comment by ahoiroman** [comment]
It is.


```
<?php

namespace App\Support\EventSourcing\Projections;

use Spatie\EventSourcing\Projections\Projection as BaseProjection;

abstract class Projection extends BaseProjection
{
    protected $guarded = [];

    public function getKeyName(): string
    {
        return 'id';
    }

    public function getRouteKeyName(): string
    {
        return 'id';
    }
}
```


And extend your projections from this.
